common/src/interlocked_hl.c

```diff
@@ -107,10 +107,19 @@ IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_WaitForValue,
 
             /* Codes_SRS_INTERLOCKED_HL_01_004: [ If the value at address is not equal to value, InterlockedHL_WaitForValue shall wait until the value at address changes in order to compare it again to value by using wait_on_address. ]*/
             /* Codes_SRS_INTERLOCKED_HL_01_005: [ When waiting for the value at address to change, the milliseconds argument value shall be used as timeout. ]*/
-            if (wait_on_address(address, current_value, milliseconds) != WAIT_ON_ADDRESS_OK)
+            WAIT_ON_ADDRESS_RESULT wait_result = wait_on_address(address, current_value, milliseconds);
+            if (wait_result != WAIT_ON_ADDRESS_OK)
             {
-                /* Codes_SRS_INTERLOCKED_HL_01_006: [ If wait_on_address fails, InterlockedHL_WaitForValue shall fail and return INTERLOCKED_HL_ERROR. ]*/
-                result = INTERLOCKED_HL_ERROR;
+                if (wait_result == WAIT_ON_ADDRESS_TIMEOUT)
+                {
+                    /* Codes_SRS_INTERLOCKED_HL_11_001: [ If wait_on_address timesout, InterlockedHL_WaitForValue shall fail and return INTERLOCKED_HL_TIMEOUT. ] */
+                    result = INTERLOCKED_HL_TIMEOUT;
+                }
+                else
+                {
+                    /* Codes_SRS_INTERLOCKED_HL_01_006: [ If wait_on_address fails, InterlockedHL_WaitForValue shall fail and return INTERLOCKED_HL_ERROR. ]*/
+                    result = INTERLOCKED_HL_ERROR;
```

Comment:
>INTERLOCKED_HL_ERROR [](http://example.com/codeflow?start=29&length=20)

Line 171 Logs an error here, probably we should log something here too.

After all - I believe we'll never see it, but for those cases when we'd potentially-virtually-allegedly see it - we'd better see it.

Reply:
I can make that happen.